tests/kernel/sched/deadline/src/main.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <ztest.h>
+
+#define NUM_THREADS 8
+#define STACK_SIZE (256 + CONFIG_TEST_EXTRA_STACKSIZE)
+
+struct k_thread worker_threads[NUM_THREADS];
+
+K_THREAD_STACK_ARRAY_DEFINE(worker_stacks, NUM_THREADS, STACK_SIZE);
+
+int thread_deadlines[NUM_THREADS];
+
+/* The number of worker threads that ran, and and array of their
+ * indices in execution order
+ */
+int n_exec;
+int exec_order[NUM_THREADS];
+
+void worker(void *p1, void *p2, void *p3)
+{
+	int tidx = (int) p1;
+
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	zassert_true(tidx >= 0 && tidx < NUM_THREADS, "");
+	zassert_true(n_exec >= 0 && n_exec < NUM_THREADS, "");
+
+	exec_order[n_exec++] = tidx;
```

Comment:
Shouldn't `n_exec++` be an atomic increment?

Reply:
The worker threads are designed to serialize, assuming correct scheduler behavior.  If they do race (in a non-trivial way), that will show up in either a miscount in n_exec or a misordered exec_order list, both of which are assertions tested.